addon/ng2/blueprints/component/index.js

```diff
@@ -16,6 +27,11 @@ module.exports = {
     var parsedPath = dynamicPathParser(this.project, entityName);
 
     this.dynamicPath = parsedPath;
+    
+    if (!validateName(parsedPath.name)) {
+      throw 'Invalid name. Names must be camelCase or kabob-case';
```

Comment:
I should know better than to trust my spelling, ever since 3rd grade when I spelled enough "enuff"

Thanks for catching that @gkalpak!